pkg/cdn/metadata/mem/mem.go

```diff
@@ -0,0 +1,36 @@
+package mem
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/spf13/afero"
+
+	"github.com/gomods/athens/pkg/cdn/metadata"
+	"github.com/gomods/athens/pkg/cdn/metadata/fs"
+	"github.com/gomods/athens/pkg/errors"
+)
+
+var (
+	l          sync.Mutex
+	memStorage metadata.Storage
+)
+
+// NewStorage creates new in-memory CDN metadata using the afero.NewMemMapFs() in memory file system
+func NewStorage() (metadata.Storage, error) {
+	const op errors.Op = "mem.NewStorage"
+	l.Lock()
+	defer l.Unlock()
+
+	if memStorage != nil {
+		return memStorage, nil
+	}
+
+	memFs := afero.NewMemMapFs()
+	tmpDir, err := afero.TempDir(memFs, "", "")
+	if err != nil {
+		return nil, errors.E(op, fmt.Errorf("could not create temp dir for 'In Memory' CDN metadata (%s)", err))
+	}
+
+	return fs.NewStorage(memFs, tmpDir), nil
```

Comment:
Please set the pkg level memStorage variable to the result of fs.NewStorage(...) before the return. Otherwise each call to mem.NewStorage will create a new instance and the behaviour will differ from other storages.